src/java/org/apache/cassandra/db/RowIndexEntry.java

```diff
@@ -343,6 +349,47 @@ public static void skipForCache(DataInputPlus in) throws IOException
             }
         }
 
+        private void checkSize(int entries, int bytes)
+        {
+            ReadCommand command = ReadCommand.getCommand();
+            if (command == null || SchemaConstants.isSystemKeyspace(command.metadata().keyspace) || !DatabaseDescriptor.getClientTrackWarningsEnabled())
+                return;
+
+            int warnThreshold = DatabaseDescriptor.getRowIndexSizeWarningThresholdKb() * 1024;
+            int abortThreshold = DatabaseDescriptor.getRowIndexSizeAbortThresholdKb() * 1024;
+
+            long estimatedMemory = estimateMaterializedIndexSize(entries, bytes);
+            ColumnFamilyStore cfs = Schema.instance.getColumnFamilyStoreInstance(command.metadata().id);
+            if (cfs != null)
+                cfs.metric.rowIndexSize.update(estimatedMemory);
+
+            if (abortThreshold != 0 && estimatedMemory > abortThreshold)
+            {
+                String msg = String.format("Query %s attempted to access a large RowIndexEntry estimated to be %d bytes " +
+                                           "in-memory (total entries: %d, total bytes: %d) but the max allowed is %d;" +
+                                           " query aborted  (see row_index_size_abort_threshold_kb)",
+                                           command.toCQLString(), estimatedMemory, entries, bytes, abortThreshold);
+                MessageParams.remove(ParamType.ROW_INDEX_ENTRY_TOO_LARGE_WARNING);
```

Comment:
...because we will have probably already hit the warning condition and we don't need to report both?

Reply:
thats the pattern we have gone with; abort is required to be larger than the warn, so why include a warn and a abort?  Only doing abort is less spam for the user IMO.

Reply:
abort adds a ClientWarn, so the question is do we return 1 or 2 logs?  I went with 1 as the abort is larger than the warn.